src/main/java/ui/UI.java

```diff
@@ -174,6 +180,12 @@ private void showMainWindow(String repoId) {
         // Should only be called after panels have been initialized
         ensureSelectedPanelHasFocus();
         initialisePickers();
+        //Startup board testing required only on app launch.
+        //Checks if the number of repos is 0 indicative of first time users of the app.
+        if (shouldCreateSampleBoardOnAppLaunch &&
+                (!TestController.isTestMode() || TestController.shouldTestStartupBoard())){
```

Comment:
Could this be factored out into a variable and named? Would be a bit easier to read.